lib/amo-client.js

```diff
@@ -110,7 +110,12 @@ Client.prototype.waitForSignedAddon = function(statusUrl, opt) {
     function checkSignedStatus() {
       self.get({url: statusUrl}).then(function(result) {
         var data = result[1];
-        if (data.processed) {
+
+        var failedValidation = !data.valid;
+        // The add-on passes validation and all files have been created.
+        var passedValidation = data.valid && data.reviewed;
```

Comment:
ugh, I think we'll just need to iterate and revisit this. Currently there is a timeout on polling so maybe I'll give that exception a little more info about the current state.